core/src/io/anuke/mindustry/core/Logic.java

```diff
@@ -150,7 +150,9 @@ public void update(){
                 Time.update();
 
                 if(state.rules.waves && state.rules.waveTimer && !state.gameOver){
-                    state.wavetime = Math.max(state.wavetime - Time.delta(), 0);
+                    if ( (state.rules.waitForWaveToEnd && (unitGroups[waveTeam.ordinal()].size() == 0)) || (!state.rules.waitForWaveToEnd)) {
```

Comment:
Can be simplified to `!state.rules.waitForWaveToEnd || unitGroups[waveTeam.ordinal()].size() == 0`. Also has unnecessary spaces before and after first parenthesis.